rclcpp_action/include/rclcpp_action/server_goal_handle.hpp

```diff
@@ -15,41 +15,250 @@
 #ifndef RCLCPP_ACTION__SERVER_GOAL_HANDLE_HPP_
 #define RCLCPP_ACTION__SERVER_GOAL_HANDLE_HPP_
 
+#include <rcl_action/types.h>
 #include <rcl_action/goal_handle.h>
 
+#include <action_msgs/msg/goal_status.hpp>
+
 #include <functional>
 #include <memory>
 
 #include "rclcpp_action/visibility_control.hpp"
 
 namespace rclcpp_action
 {
-template<typename ACTION>
-class ServerGoalHandle
+
+/// Base class to interact with goals on a server.
+/// \internal
+/**
+ *
+ * This class in not be used directly by users writing an action server.
+ * Instead users will be given an instance of `rclcpp_action::ServerGoalHandle<>`.
+ *
+ * Internally, this class is responsible for interfacing with the `rcl_action` API.
+ */
+class ServerGoalHandleBase
 {
 public:
-  virtual ~ServerGoalHandle();
-
   /// Indicate if client has requested this goal be cancelled.
   /// \return true if a cancelation request has been accepted for this goal.
-  virtual bool
-  is_cancel_request() const = 0;
+  RCLCPP_ACTION_PUBLIC
+  bool
+  is_cancel_request() const;
```

Comment:
Just curious, why did you name it like this, rather than say something like `is_canceled()`?

Reply:
It returns `true` in state `CANCELLING` but `false` in state `CANCELED`. Returning `false` would be a strange case if the API was  called `is_canceled()`. 

Ideas? It's the API an action server author should check and call `set_canceled()` after they see this returns `true`.

Alternatively, it could be called `is_canceled()` and return `true` in both cases.

Reply:
changed to `is_canceling()` in 52ac6d0bb493e709ffbb678db62a4d8c2a501d3c